src/con_duct/__main__.py

```diff
@@ -162,21 +162,36 @@ def prepare_paths(self, clobber: bool, capture_outputs: Outputs) -> None:
 
 @dataclass
 class Averages:
-    rss: float = 0.0
-    vsz: float = 0.0
-    pmem: float = 0.0
-    pcpu: float = 0.0
+    rss: Optional[float] = None
+    vsz: Optional[float] = None
+    pmem: Optional[float] = None
+    pcpu: Optional[float] = None
     num_samples: int = 0
 
     def update(self: Averages, other: Sample) -> None:
         assert_num(
             other.total_rss, other.total_vsz, other.total_pmem, other.total_pcpu
         )
-        self.num_samples += 1
-        self.rss += (other.total_rss - self.rss) / self.num_samples
-        self.vsz += (other.total_vsz - self.vsz) / self.num_samples
-        self.pmem += (other.total_pmem - self.pmem) / self.num_samples
-        self.pcpu += (other.total_pcpu - self.pcpu) / self.num_samples
+        if not self.num_samples:
+            self.num_samples += 1
+            self.rss = other.total_rss
+            self.vsz = other.total_vsz
+            self.pmem = other.total_pmem
+            self.pcpu = other.total_pcpu
+        else:
+            assert self.rss is not None
+            assert self.vsz is not None
+            assert self.pmem is not None
+            assert self.pcpu is not None
+            assert other.total_rss is not None
+            assert other.total_vsz is not None
+            assert other.total_pmem is not None
+            assert other.total_pcpu is not None
```

Comment:
just a note since not relevant to this PR: I have odd feeling here about code that this `self` is having assumptions about `other` (Sample)